### ISSUE TITLE
Typo in JavaScript Learning Exercise: Incorrect Unit Mentioned in Lesson Instructions

### ISSUE THREAD

**Issue by iQadeerAhmad** [freeCodeCamp/freeCodeCamp#51611]
### Describe the Issue

Hello FreeCodeCamp team,

I found a minor typo in the JavaScript learning module. In the lesson titled "Use getters and setters to Control Access to an Object," there is a small error in the instruction related to creating a getter and setter for temperature conversion. Here's the specific portion of the instruction that needs correction:

**Incorrect Line:**
"In the class, create a getter to obtain the temperature in Celsius and a setter to set the temperature in Celsius."

**Suggested Correction:**
"In the class, create a getter to obtain the temperature in Celsius and a setter to set the temperature in Fahrenheit."

This correction is necessary to maintain clarity and accuracy in the learning material. The correct units (Celsius and Fahrenheit) should align with the given problem statement and the intended outcome of the exercise.

Thank you for your attention to detail and dedication to improving the learning experience on FreeCodeCamp.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/es6/use-getters-and-setters-to-control-access-to-an-object

### Steps to Reproduce

Navigate to the javascript algorithms and data structures learning module.
Go to ES6 course.
Locate the lesson titled "Use getters and setters to Control Access to an Object."
Read the relevant portion of the instruction mentioned.

### Recommended fix or suggestions

To correct the mentioned typo and ensure clarity in the instruction, the line should be modified as follows:

**Original:**
"In the class, create a getter to obtain the temperature in Celsius and a setter to set the temperature in Celsius."

**Suggested Correction:**
"In the class, create a getter to obtain the temperature in Celsius and a setter to set the temperature in Fahrenheit."

This change aligns with the problem statement and maintains consistency with the intended temperature units.

### Screenshots

![Screenshot from 2023-09-20 01-32-01](https://github.com/freeCodeCamp/freeCodeCamp/assets/78398774/58103660-2fb6-4394-b1d2-4434e509624c)

**Comment by moT01** [comment]
I'm not sure I see the problem here. How do we know that the intent is for the setter to set the temp in Fahrenheit? Maybe I'm missing something. The seed code seems to indicate that it should be set in Celsius:

```
const thermos = new Thermostat(76); // Setting in Fahrenheit scale
let temp = thermos.temperature; // 24.44 in Celsius
thermos.temperature = 26; // *** Here it gets set in Celsius
temp = thermos.temperature; // 26 in Celsius *** And here it outputs the same number in Celsius
```

**Comment by iQadeerAhmad** [comment]
no, in the last line

`temp = thermos.temperature; // 26 in Celsius` *** It also runs the getter function, which means that it takes Celsius as Fahrenheit and tries to convert it into Celsius that's why we have to set the temperature in Fahrenheit. Here is the working code.


`class Thermostat{
  constructor(Fahrenheit){
    this._Fahrenheit=Fahrenheit;
  }
  get temperature(){
    return (5/9*(this._Fahrenheit-32));
  }
  
  set temperature(celsius){
    this._Fahrenheit = celsius * 9.0 / 5 + 32;
  }
}`

**Comment by bbsmooth** [comment]
Ya, I'm not seeing a problem here either. The getter spits out a Celsius temp and the setter takes a Celsius temp. 

How you actually store the temp internally in the class is not a consideration here. You don't have to store the internal temp in Fahrenheit. You can solve this by storing the temp internally in Celsius:

```
class Thermostat{
  constructor(Fahrenheit){
    this._temp=5/9*(Fahrenheit-32);
  }
  get temperature(){
    return this._temp;
  }
  set temperature(celsius){
    this._temp = celsius;
  }
}
```

**Comment by iQadeerAhmad** [comment]
Yes, I agree with you, we can solve this by storing temp internally. 

**"The constructor accepts a Fahrenheit temperature. In the class, create a getter to obtain the temperature in Celsius and a setter to set the temperature in Celsius."**

I could be wrong but the above lines clearly mention that we have to use the getter function to get the temperature in Celsius and the constructor has to accept temperature in Fahrenheit. I feel that these lines are a bit confusing.

**Comment by moT01** [comment]
I guess I can see how it could be confusing - but I don't think your suggestion here is the way to go. It's setting the temp in Celsius: `thermos.temperature = 26;` So saying that the setter function should set the temp in Fahrenheit doesn't seem to make sense to me. What about `create a getter to obtain the temperature in Celsius and a setter that accepts a temperature in Celsius` or something.

**Comment by iQadeerAhmad** [comment]
> I guess I can see how it could be confusing - but I don't think your suggestion here is the way to go. It's setting the temp in Celsius: `thermos.temperature = 26;` So saying that the setter function should set the temp in Fahrenheit doesn't seem to make sense to me. What about `create a getter to obtain the temperature in Celsius and a setter that accepts a temperature in Celsius` or something.

Yes, we have to make this line more clear so students won't get confused.

**Comment by RockyPukar1** [comment]
`let temp = thermos.temperature`
This line makes a requirement to use temperature as a parameter for the constructor and also the name of the getter and setter function to be temperature.

So, for better user understanding we can user this line:
`Use the class keyword to create a Thermostat class. The constructor accepts a Fahrenheit temperature.`
like this:
`Use the class keyword to create a Thermostat class. The constructor accepts a Fahrenheit temperature. Use temperature as a parameter for constructor`.
So that final code becomes like this:

```
// Only change code below this line
class Thermostat {
  constructor(temperature) {
    this._temperature = temperature;
  }

  get temperature() {
    return 5/9 * (this._temperature - 32);
  }

  set temperature(celsius) {
    this._temperature = celsius * 9.0 / 5 + 32
  }
}
// Only change code above this line

const thermos = new Thermostat(76); // Setting in Fahrenheit scale
let temp = thermos.temperature; // 24.44 in Celsius
thermos.temperature = 26;
temp = thermos.temperature; // 26 in Celsius
```

**Comment by RockyPukar1** [comment]
Also the assert functions also requires temperature to be set
```
const _desc = Object.getOwnPropertyDescriptor(Thermostat.prototype, 'temperature');
assert.isFunction(_desc?.get);
```
```
const _desc = Object.getOwnPropertyDescriptor(Thermostat.prototype, 'temperature');
assert.isFunction(_desc?.set);
```

**Comment by RockyPukar1** [comment]
and standard says the getter and setter function should have the same name as the property with _ removed for which property the getter and setter are working

**Comment by jeremylt** [comment]
I think 

>In the class, create a `getter` to obtain the temperature in Celsius and a `setter`that accepts a temperature in Celsius.

is the way to go.

We do not need to specify the parameter name of the constructor.

**Comment by RockyPukar1** [comment]
@jeremylt Yes I agree with you. Knowing what to put in the parameter in the constructor looking
`let temp = thermos.temperature`
will be a task that requires good understanding for the students.

**Comment by moT01** [comment]
I added the first timer label. The [file that needs to be changed is here.](https://github.com/freeCodeCamp/freeCodeCamp/blob/f4c4f9ac30507670f9ca8cf679b39d689059d1af/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-getters-and-setters-to-control-access-to-an-object.md?plain=1#L49) We want to change it to the suggestion:

> In the class, create a `getter` to obtain the temperature in Celsius and a `setter` that accepts a temperature in Celsius.

If you would like to contribute, and have not read the contributors docs, please do so here: https://contribute.freecodecamp.org/#/

**Comment by surajkewat** [comment]
#51658 I have raised PR for change
